src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -1399,25 +1399,43 @@ bool NetworkOPsImp::checkLastClosedLedger (
 }
 
 void NetworkOPsImp::switchLastClosedLedger (
-    Ledger::pointer newLedger, bool duringConsensus)
+    Ledger::pointer newLCL, bool duringConsensus)
 {
-    // set the newledger as our last closed ledger -- this is abnormal code
+    // set the newLCL as our last closed ledger -- this is abnormal code
 
     auto msg = duringConsensus ? "JUMPdc" : "JUMP";
-    m_journal.error << msg << " last closed ledger to " << newLedger->getHash ();
+    m_journal.error << msg << " last closed ledger to " << newLCL->getHash ();
 
     clearNeedNetworkLedger ();
-    newLedger->setClosed ();
-    auto openLedger = std::make_shared<Ledger> (false, std::ref (*newLedger));
-    m_ledgerMaster.switchLedgers (newLedger, openLedger);
+    newLCL->setClosed ();
+    auto newOL = std::make_shared<Ledger> (false, std::ref (*newLCL));
+    // Caller must own master lock
```

Comment:
The only way I can think of doing this is to demand that the client use `unique_lock`, and have them pass it in by const&, then here we can `assert(lk.owns_lock() && lk.mutex() == &app().getMasterMutex())`.